llpc/context/llpcCompiler.cpp

```diff
@@ -777,8 +776,12 @@ Result Compiler::BuildShaderModule(const ShaderModuleBuildInfo *shaderInfo, Shad
 // @param context : Acquired context
 // @param shaderInfo : Shader info of this pipeline
 // @param [out] pipelineElf : Output Elf package
+// @param [out] stageCacheAccesses : Stage cache access result. All elements
+//                                   must be initialized in the caller as
+//                                   StageCacheInfo::CacheNotChecked
 Result Compiler::buildPipelineWithRelocatableElf(Context *context, ArrayRef<const PipelineShaderInfo *> shaderInfo,
-                                                 ElfPackage *pipelineElf) {
+                                                 ElfPackage *pipelineElf,
+                                                 StageCacheInfo stageCacheAccesses[ShaderStageCount]) {
```

Comment:
Don't use C arrays as function arguments. Prefer `llvm::ArrayRef`, or `llvm::MutableArrayRef` when the size is static but not known, `std::array<...>` when the size is static and known, or `llvm::SmallVectorImpl` when the size is dynamic.